### PR TITLE
chore(#1517): Reshuffle development workflow to match dev journey

### LINKED ISSUE #1517: Reorganize the sections in the development workflow page
The current [Development workflow page](https://docs.communityhealthtoolkit.org/contribute/code/workflow/) sections do not follow the logical journey of the development workflow. For example, the `Issues` section should probably be before the `Branches` and `Commits` ones. Also, `Updating The Issue With What You Actually Did` should move before the `Code Reviews `section.

### PR DESCRIPTION
**Pull request by andrablaj**
# Description

Closes #1517 

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: content/en/contribute/code/workflow.md [hunk 1]
```diff
@@ -26,6 +97,31 @@ A good workflow would be to work locally, pushing to a remote branch as you make
...
+   
+   - Coordinate with a QA engineer and add them as reviewers when you need specific quality/testing support. For example, when major changes or new features are introduced to the codebase, security-related changes are made, and substantial user experience improvements are required. The QA engineer can advise whether that needs to be considered in the current e2e suite, they can advise on edge cases or other scenarios to consider for testing.
+   
+6. Follow up on your PR to keep momentum; the review should happen in 24h business days. If you haven't received feedback from the reviewers after that time, check if they are available. Otherwise, it's okay to request a review from another person.
+7. Once the PR has been approved by a QA engineer and a developer, wait for the GitHub Actions to succeed and ensure there are no conflicts with the main branch.
+8. Double-check the [commit message format]({{< ref "#commit-message-format" >}}) is correct. Make sure to recognize collaboration in the commit description: `Co-authored-by: <gitnub user>`.
```
Comment (latin-panda) [inline content/en/contribute/code/workflow.md:121]:
```suggestion
8. Double-check the [commit message format]({{< ref "#commit-message-format" >}}) is correct. Make sure to recognize collaboration in the commit description: `Co-authored-by: <GitHub user>`.
```

File: content/en/contribute/code/workflow.md [hunk 2]
```diff
@@ -26,6 +97,31 @@ A good workflow would be to work locally, pushing to a remote branch as you make
...
+9. Merge your work by selecting `Squash and merge`. This will compress all the commits into one, keeping the repository's commit history clean.
+   - If a backport is required, cherry-pick the merged commit back to the release branches it is required in.
+
```
Comment (latin-panda) [inline content/en/contribute/code/workflow.md:124]: L133 seems kinda similar to L139 and can be merged into one

File: content/en/contribute/code/workflow.md [hunk 3]
```diff
@@ -57,43 +153,6 @@ Timely code reviews are important to getting improvements into the hands of user
 
 Code reviews should be completed within 24 hours of assignment (excluding weekends and holidays). In some cases, a code review may not be possible if a larger discussion needs to be had for design choices or solution objectives, but even in cases like those, some feedback is still to be expected within 24 hours.
```
Comment (latin-panda) [inline content/en/contribute/code/workflow.md:154]: Some of this already mentioned in L112

**Review by latin-panda — APPROVED** [review]
Great improvement! Thank you

Just minor suggestions, approving now

File: content/en/contribute/code/workflow.md (hunk 3 above)
Comment (andrablaj) [reply content/en/contribute/code/workflow.md:154]: I would keep the two mentions to ensure people are clear on the 24-hour timeliness mark.

**Comment by esthermmoturi** [comment]
LGTM! Thanks @andrablaj.

**Review by esthermmoturi — APPROVED** [review]

**Pull request merged**